src/core/qgsapplication.cpp

```diff
@@ -1434,22 +1440,33 @@ int QgsApplication::systemMemorySizeMb()
   memoryStatus.dwLength = sizeof( MEMORYSTATUSEX );
   if ( GlobalMemoryStatusEx( &memoryStatus ) )
   {
-    return memoryStatus.ullTotalPhys / ( 1024 * 1024 );
+    return memoryStatus.ullTotalPhys / megabyte;
   }
   else
   {
     return -1;
   }
 #elif defined(Q_OS_LINUX)
-  constexpr int megabyte = 1024 * 1024;
   struct sysinfo si;
   sysinfo( &si );
   return si.totalram / megabyte;
-#elif defined(Q_OS_FREEBSD)
-  return -1;
-#elif defined(Q_OS_OPENBSD)
-  return -1;
-#elif defined(Q_OS_NETBSD)
+#elif defined(Q_OS_FREEBSD) || defined(Q_OS_OPENBSD) || defined(Q_OS_NETBSD)
+  unsigned long physmem;
+  size_t size = sizeof( physmem );
```

Comment:
On NetBSD, sysctl(7) says that hw.physmem64  is a "64-bit integer'.   That type would be int64_t, not unsigned long.
And hy.physmem is a "32-bit integer", so "int32_t".